x/slashing/client/rest/query.go

```diff
@@ -2,44 +2,84 @@ package rest
 
 import (
 	"fmt"
-	"net/http"
-
-	"github.com/gorilla/mux"
-
 	"github.com/cosmos/cosmos-sdk/client/context"
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/rest"
 	"github.com/cosmos/cosmos-sdk/x/slashing"
+	"github.com/gorilla/mux"
+	"net/http"
 )
 
 func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *codec.Codec) {
 	r.HandleFunc(
-		"/slashing/validators/{validatorPubKey}/signing_info",
+		"/slashing/validators/signing_info/{validatorPubKey}",
 		signingInfoHandlerFn(cliCtx, slashing.StoreKey, cdc),
 	).Methods("GET")
 
+	r.HandleFunc(
+		"/slashing/validators/signing_info",
+		signingInfoHandlerListFn(cliCtx, slashing.StoreKey, cdc),
+	).
+		Methods("GET").
+		Queries("page", "{page}", "limit", "{limit}")
+
+	r.HandleFunc(
+		"/slashing/validators/signing_info",
```

Comment:
Why is this route defined three times? Can we not have all the query params defined/used once?

Reply:
I've seen gorilla can handle queries in this way, it looks weird to me but seems if they are not registered many times they won't match:
https://stackoverflow.com/questions/45378566/gorilla-mux-optional-query-values

Reply:
Ahhh I see. No gusta. How about your thoughts on just having:

```
r.HandleFunc(
		"/slashing/validators/signing_info",
		signingInfoHandlerListFn(cliCtx, slashing.StoreKey, cdc),
	).Methods("GET").Queries("page", "{page}", "limit", "{limit}")
```

Thus we require pagination.

Reply:
All or nothing! sounds good, I was leaving some choice but seems healthier to force clients to think about this detail initially